server/src/web/user/user.controller.ts

```diff
@@ -0,0 +1,46 @@
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    Inject,
+    Injectable,
+    Param, ParseIntPipe,
+    Post, Put,
+    Query,
+} from '@nestjs/common';
+import {User} from '../../domain/user/user.entity';
+import {UserDto} from '../../app/user/dto/user.dto';
+import {DeleteResult} from 'typeorm';
+import {UserService} from '../../app/user/user.service';
+
+@Injectable()
+@Controller('users')
+export class UserController {
+    constructor(@Inject('UserService') private service: UserService) {}
+
+    @Get(':id')
+    async get(@Param('id') id: number): Promise<User> {
+        return await this.service.get(id);
+    }
+    @Post()
+    async create(@Body() userDto: UserDto): Promise<User> {
+        return await this.service.create(userDto);
+    }
+    @Delete(':id')
+    async delete(@Param('id') id: number): Promise<DeleteResult> {
+        return await this.service.delete(id);
+    }
+    @Put(':id/increase-level')
+    async increaseLevel(@Param('id') id: number, @Query('amount', new ParseIntPipe()) amount): Promise<User> {
+        return await this.service.increaseLevel(id, amount);
+    }
+    @Put(':id/increase-point')
```

Comment:
Also, I suggest that it changes to `@Put(':id/point')` and `amount` has positive amount or negative amount.

Reply:
point is not object. that's why i didn't use noun for point at the url

Reply:
How about `@Put(':id?point=amount')`?